static/specification.html

```diff
@@ -138,605 +336,555 @@ <h2>
+            </p>
+          </li>
+          <li>
+            <p>
+              If <var>element</var>'s <span>node document</span> is not
+              [=allowed to use=] the "monetization" feature, return false.
+            </p>
+          </li>
+          <li>
+            <p>
+              Let <var>context</var> be <var>element</var>'s <span>node
+              document</span>'s <span data-x="concept-document-bc">browsing
+              context</span>.
+            </p>
+          </li>
+          <li>
+            <p>
+              Set <var>request</var>'s <span data-x=
+              "concept-request-initiator">initiator</span> to "`document`".
+            </p>
+          </li>
+          <li>
+            <p>
+              Set <var>request</var>'s <span data-x=
+              "concept-request-destination">destination</span> to
+              "<code data-x="">monetization</code>".
+            </p>
+          </li>
+          <li>
+            <p>
+              Set <var>request</var>'s <span data-x=
+              "concept-request-mode">mode</span> to "<code data-x=
+              "">cors</code>".
+            </p>
+          </li>
+          <li>
+            <p>
+              Set <var>request</var>'s <span data-x=
+              "concept-request-credentials-mode">credentials mode</span> to the
+              <span>CORS settings attribute credentials mode</span> for
+              <var>element</var>'s <code data-x=
+              "attr-link-crossorigin">crossorigin</code> content attribute.
+            </p>
+          </li>
+          <li>
+            <p>
+              Return true.
+            </p>
+          </li>
+        </ol>
+        <p>
+          To <span data-x="process the linked resource">process this type of
+          linked resource</span> given a <code>link</code> element
+          <var>element</var>, boolean <var>success</var>, and <span data-x=
+          "concept-response">response</span> <var>response</var>:
+        </p>
+        <ol>
+          <li>
+            <p>
+              If |response|'s status is not an OK status, the set |success| to
+              false.
+            </p>
+          </li>
+          <li>
+            <p>
+              Otherwise, if <var>response</var>'s <span data-x=
+              "Content-Type">Content-Type metadata</span> is not a
+              <code>application/json</code>, then set <var>success</var> to
+              false.
+            </p>
+          </li>
+          <li>Let |json| be the result of [=parse JSON bytes to an Infra
+          value=] passing |response|'s [=Response/body=].
+          </li>
+          <li>If |json| is an exception, then set <var>success</var> to false.
+          </li>
+          <li>
+            <p>
+              If the user agent has exhausted the number of allowed [=payment
+              sessions=], set |success| to false.
+            </p>
+          </li>
+          <li>Otherwise, establish a new [=payment session=].
```

Comment:
Some breakdown here:
https://webmonetization-preview.netlify.app/docs/web-monetization-api#web-monetization-concepts-and-usage

Reply:
> The Web Monetization Agent is designed to be implemented in browsers and expose the Web Monetization API. It's responsible to:
get the rate, token, and payment connection address from the Web Monetization Provider

How will tips be supported in this model? 
How is the rate calculated ? It's a flat rate? If so, the backend could set this. 
Would the provider need access to the active tab' url to create a dynamic rate?

Reply:
Only the browser implements this spec. Its role is to provide an interface for websites to code against.

The browser is responsible for fetching the payment pointer URL and doing any processing necessary to pass payment details to the WM provider without leaking any details of the website that has been visited or the payment pointer. 

Following that, the website emits a load event so the website knows that WM is enabled and the payment pointer was successfully processed. What the website does with that info is up to it. There is no guarantee that any payments will follow or if they do, how often, or how large they will be.

The last role of the browser is to emit an event each time a payment is sent. The event should provide enough data for the website to verify the payment independently since the event is simply a notification that there was an attempt to pay. Again, depending on the context, the website may do different things with this information. For micropayments in exchange for streaming content the website may not bother verifying all payments for larger payments in exchange for some good or service the website might verify the payment before delivering the good or service.

The WM Provider is responsible for executing payments using the details provided by the browser. The WM Provider defines how much to send and when either through some heuristics or user input (e.g. tipping).

The interface between the browser and WM Provider is NOT defined here. This is intentional so that we leave flexibility in how this might evolve in future. If we hadn't leaked the receipt logic out through this API we might have been able to easily switch between SPSP and OP without any changes :(